### =========================================================================
### excluderanges metadata
### -------------------------------------------------------------------------
###

# Empty data frame
# meta <- data.frame(
#     Title = c(),
#     Description = c(),
#     BiocVersion = rep(),
#     Genome = c(),
#     SourceType = rep(),
#     SourceUrl = rep(),
#     SourceVersion = rep(),
#     Species = c(),
#     TaxonomyId = c(),
#     Coordinate_1_based = c(),
#     DataProvider = c(),
#     Maintainer = rep(),
#     RDataClass = rep(),
#     DispatchClass = rep(),
#     RDataPath = c(),
#     Tags = c(),
#     Notes = "")

# Complete data frame
meta <- data.frame(
  Title = c("T2T.excluderanges.rds",
            "hg38.Kundaje.GRCh38_unified_Excludable.rds",
            "hg38.Bernstein.Mint_Excludable_GRCh38.rds",
            "hg38.Boyle.hg38-Excludable.v2.rds",
            "hg38.Kundaje.GRCh38.Excludable.rds",
            "hg38.Lareau.hg38.full.Excludable.rds",
            "hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds",
            "hg38.Wimberley.peakPass60Perc_sorted.rds",
            "hg38.Wold.hg38mitoExcludable.rds",
            "hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds",
            "hg19.Boyle.hg19-Excludable.v2.rds",
            "hg19.Bernstein.Mint_Excludable_hg19.rds",
            "hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds",
            "hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds",
            "hg19.Lareau.hg19.full.Excludable.rds",
            "hg19.Wold.hg19mitoExcludable.rds",
            "hg19.Yeo.eCLIP_Excludableregions.hg19.rds",
            "mm39.excluderanges.rds",
            "mm10.Boyle.mm10-Excludable.v2.rds",
            "mm10.Hardison.Excludable.full.rds",
            "mm10.Hardison.psuExcludable.mm10.rds",
            "mm10.Kundaje.anshul.Excludable.mm10.rds",
            "mm10.Kundaje.mm10.Excludable.rds",
            "mm10.Lareau.mm10.full.Excludable.rds",
            "mm10.Wold.mm10mitoExcludable.rds",
            "mm9.Lareau.mm9.full.Excludable.rds",
            "mm9.Wold.mm9mitoExcludable.rds",
            "ce11.Boyle.ce11-Excludable.v2.rds",
            "ce10.Boyle.ce10-Excludable.v2.rds",
            "ce10.Kundaje.ce10-Excludable.rds",
            "danRer10.Domingues.Excludableed.rds",
            "danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome.rds",
            "dm6.Boyle.dm6-Excludable.v2.rds",
            "dm3.Boyle.dm3-Excludable.v2.rds",
            "dm3.Kundaje.dm3-Excludable.rds",
            "TAIR10.Wimberley.predicted_excluded_list_sorted_0.6.rds",
            "TAIR10.Klasfeld.arabidopsis_Excludable_20inputs.rds",
            "TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs.rds",
            "T2T.Lareau.chm13v2.0_peaks.rds",
            "hg38.Lareau.hg38_peaks.rds",
            "hg19.Lareau.hg19_peaks.rds",
            "mm10.Lareau.mm10_peaks.rds",
            "mm9.Lareau.mm9_peaks.rds",
            "hg19.UCSC.numtS.rds",
            "mm9.UCSC.numtS.rds",
            "T2T.CHM13.chm13.draft_v2.0.cen_mask.rds",
            "T2T.CHM13.chm13.draft_v1.1.telomere.rds",
            "T2T.UCSC.censat.rds",
            "T2T.UCSC.gap.rds",
            "T2T.UCSC.hgUnique.hg38.rds",
            "hg38.UCSC.centromere.rds",
            "hg38.UCSC.telomere.rds",
            "hg38.UCSC.short_arm.rds",
            "hg38.UCSC.heterochromatin.rds",
            "hg38.UCSC.contig.rds",
            "hg38.UCSC.scaffold.rds",
            "hg19.UCSC.centromere.rds",
            "hg19.UCSC.telomere.rds",
            "hg19.UCSC.short_arm.rds",
            "hg19.UCSC.heterochromatin.rds",
            "hg19.UCSC.clone.rds",
            "hg19.UCSC.contig.rds",
            "hg19.UCSC.scaffold.rds",
            "mm39.UCSC.centromere.rds",
            "mm39.UCSC.telomere.rds",
            "mm39.UCSC.short_arm.rds",
            "mm39.UCSC.contig.rds",
            "mm39.UCSC.scaffold.rds",
            "mm10.UCSC.centromere.rds",
            "mm10.UCSC.telomere.rds",
            "mm10.UCSC.short_arm.rds",
            "mm10.UCSC.clone.rds",
            "mm10.UCSC.contig.rds",
            "mm10.UCSC.scaffold.rds",
            "mm10.UCSC.other.rds",
            "mm10.UCSC.fragment.rds",
            "mm9.UCSC.centromere.rds",
            "mm9.UCSC.fragment.rds",
            "mm9.UCSC.contig.rds",
            "danRer10.UCSC.contig.rds",
            "danRer10.UCSC.scaffold.rds",
            "dm6.UCSC.other.rds",
            "dm3.UCSC.contig.rds",
            "dm3.UCSC.scaffold.rds",
            "TAIR10.UCSC.araTha1.gap.rds"),
  Description = c("Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined as a combination of hg38.Lareau.hg38_peaks, hg38.Boyle.hg38-Excludable.v2, and hg38.Wimberley.peakPass60Perc_sorted, followed by manual curation. Supersedes hg38.Kundaje.GRCh38.Excludable.",
                  "Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Defined by the ewimberley/peakPass software",
                  "Definition method unknown",
                  "Defined from eCLIP data",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data",
                  "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands",
                  "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), human satellite repeat HSATII, chrM, ribosomal subunit consensus sequences LSU-rRNA_Hsa, SSU-rRNA_Hsa. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Definition method unknown",
                  "Defined from eCLIP data, includes skyscraper, rRNA pseudogene, unreliably mapped satellite repeat, and low complexity skyscraper peak regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Definition method unknown",
                  "Definition method unknown",
                  "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
                  "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Definition method unknown",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Definition method unknown",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by Anshul Kundaje, superseded by ce10.Boyle.ce10-Excludable.v2",
                  "Defined manually using total RNA-seq.",
                  "Defined via MACS2 peak calling using ChIP-seq (PMID: 33239788)",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by Anshul Kundaje. Contains heterochromatin chromosomes chr2LHet. Superseded by dm3.Boyle.dm3-Excludable.v2",
                  "Defined by the ewimberley/peakPass software",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions (DOI: 10.1101/2022.02.27.482177)",
                  "Defined by the green screen pipeline (DOI: 10.1101/2022.02.27.482177)",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Human NumtS mitochondrial sequence",
                  "Mouse NumtS mitochondrial sequence",
                  "Centromeric satellite masking bed file (v2.0)",
                  "Telomere identified by the VGP pipeline (v1.1)",
                  "T2T peri/centromeric satellite annotation (v2.0, 20220329, CHM13 v2.0)",
                  "Locations of assembly gaps, as determine by strings of 'N' characters (v1.0)",
                  "Regions unique to the T2T-CHM13 v2.0 assembly compared to the GRCh38/hg38 and GRCh37/hg19 reference assemblies",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps from large blocks of heterochromatin",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps from large blocks of heterochromatin",
                  "Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies. Only non-autosomal chromosomes",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "A single gap of 31 bases in chrX_GL456233_random",
                  "Gaps from centromeres",
                  "Gaps between the contigs of a draft clone. (In this context, a contig is a set of overlapping sequence reads). Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Gaps in the May 2011 Arabidopsis thaliana genome assembly"),
  BiocVersion = rep("3.16", 85),
  Genome = c("T2T-CHM13",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "mm39",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm9",
             "mm9",
             "ce11",
             "ce10",
             "ce10",
             "danRer10",
             "danRer10",
             "dm6",
             "dm3",
             "dm3",
             "TAIR10",
             "TAIR10",
             "TAIR10",
             "T2T-CHM13",
             "hg38",
             "hg19",
             "mm10",
             "mm9",
             "hg19",
             "mm9",
             "T2T-CHM13",
             "T2T-CHM13",
             "T2T-CHM13",
             "T2T-CHM13",
             "T2T-CHM13",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "mm39",
             "mm39",
             "mm39",
             "mm39",
             "mm39",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm9",
             "mm9",
             "mm9",
             "danRer10",
             "danRer10",
             "dm6",
             "dm3",
             "dm3",
             "TAIR10"),
  SourceType = rep("RDS", 85),
  SourceUrl = rep("https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2kfHOWFBSJT?usp=sharing", 85),
  SourceVersion = rep("September 22 2022", 85),
  Species = c("Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Caenorhabditis elegans",
              "Caenorhabditis elegans",
              "Caenorhabditis elegans",
              "Danio rerio",
              "Danio rerio",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Arabidopsis thaliana",
              "Arabidopsis thaliana",
              "Arabidopsis thaliana",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Homo sapiens",
              "Mus musculus",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Danio rerio",
              "Danio rerio",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Arabidopsis thaliana"),
  TaxonomyId = c("9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "6239",
                 "6239",
                 "6239",
                 "7955",
                 "7955",
                 "7227",
                 "7227",
                 "7227",
                 "3702",
                 "3702",
                 "3702",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "9606",
                 "10090",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "7955",
                 "7955",
                 "7227",
                 "7227",
                 "7227",
                 "3702"),
  Coordinate_1_based = c(TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         FALSE,
                         FALSE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         TRUE),
  DataProvider = c("excluderanges",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "excluderanges",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "GitHub",
                   "Stanford.edu",
                   "GitHub",
                   "Publication",
                   "GitHub",
                   "GitHub",
                   "Stanford.edu",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "UCSC",
                   "UCSC",
                   "CHM13",
                   "CHM13",
                   "UCSChub",
                   "UCSChub",
                   "UCSChub",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSChub"),
  Maintainer = rep("Mikhail Dozmorov <mikhail.dozmorov@gmail.com>", 85),
  RDataClass = rep("GRanges", 85),
  DispatchClass = rep("RDS", 85),
  RDataPath = c("excluderanges/T2T.excluderanges.rds",
                "excluderanges/hg38.Kundaje.GRCh38_unified_Excludable.rds",
                "excluderanges/hg38.Bernstein.Mint_Excludable_GRCh38.rds",
                "excluderanges/hg38.Boyle.hg38-Excludable.v2.rds",
                "excluderanges/hg38.Kundaje.GRCh38.Excludable.rds",
                "excluderanges/hg38.Lareau.hg38.full.Excludable.rds",
                "excluderanges/hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds",
                "excluderanges/hg38.Wimberley.peakPass60Perc_sorted.rds",
                "excluderanges/hg38.Wold.hg38mitoExcludable.rds",
                "excluderanges/hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds",
                "excluderanges/hg19.Boyle.hg19-Excludable.v2.rds",
                "excluderanges/hg19.Bernstein.Mint_Excludable_hg19.rds",
                "excluderanges/hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds",
                "excluderanges/hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds",
                "excluderanges/hg19.Lareau.hg19.full.Excludable.rds",
                "excluderanges/hg19.Wold.hg19mitoExcludable.rds",
                "excluderanges/hg19.Yeo.eCLIP_Excludableregions.hg19.rds",
                "excluderanges/mm39.excluderanges.rds",
                "excluderanges/mm10.Boyle.mm10-Excludable.v2.rds",
                "excluderanges/mm10.Hardison.Excludable.full.rds",
                "excluderanges/mm10.Hardison.psuExcludable.mm10.rds",
                "excluderanges/mm10.Kundaje.anshul.Excludable.mm10.rds",
                "excluderanges/mm10.Kundaje.mm10.Excludable.rds",
                "excluderanges/mm10.Lareau.mm10.full.Excludable.rds",
                "excluderanges/mm10.Wold.mm10mitoExcludable.rds",
                "excluderanges/mm9.Lareau.mm9.full.Excludable.rds",
                "excluderanges/mm9.Wold.mm9mitoExcludable.rds",
                "excluderanges/ce11.Boyle.ce11-Excludable.v2.rds",
                "excluderanges/ce10.Boyle.ce10-Excludable.v2.rds",
                "excluderanges/ce10.Kundaje.ce10-Excludable.rds",
                "excluderanges/danRer10.Domingues.Excludableed.rds",
                "excluderanges/danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome.rds",
                "excluderanges/dm6.Boyle.dm6-Excludable.v2.rds",
                "excluderanges/dm3.Boyle.dm3-Excludable.v2.rds",
                "excluderanges/dm3.Kundaje.dm3-Excludable.rds",
                "excluderanges/TAIR10.Wimberley.predicted_excluded_list_sorted_0.6.rds",
                "excluderanges/TAIR10.Klasfeld.arabidopsis_Excludable_20inputs.rds",
                "excluderanges/TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs.rds",
                "excluderanges/T2T.Lareau.chm13v2.0_peaks.rds",
                "excluderanges/hg38.Lareau.hg38_peaks.rds",
                "excluderanges/hg19.Lareau.hg19_peaks.rds",
                "excluderanges/mm10.Lareau.mm10_peaks.rds",
                "excluderanges/mm9.Lareau.mm9_peaks.rds",
                "excluderanges/hg19.UCSC.numtS.rds",
                "excluderanges/mm9.UCSC.numtS.rds",
                "excluderanges/T2T.CHM13.chm13.draft_v2.0.cen_mask.rds",
                "excluderanges/T2T.CHM13.chm13.draft_v1.1.telomere.rds",
                "excluderanges/T2T.UCSC.censat.rds",
                "excluderanges/T2T.UCSC.gap.rds",
                "excluderanges/T2T.UCSC.hgUnique.hg38.rds",
                "excluderanges/hg38.UCSC.centromere.rds",
                "excluderanges/hg38.UCSC.telomere.rds",
                "excluderanges/hg38.UCSC.short_arm.rds",
                "excluderanges/hg38.UCSC.heterochromatin.rds",
                "excluderanges/hg38.UCSC.contig.rds",
                "excluderanges/hg38.UCSC.scaffold.rds",
                "excluderanges/hg19.UCSC.centromere.rds",
                "excluderanges/hg19.UCSC.telomere.rds",
                "excluderanges/hg19.UCSC.short_arm.rds",
                "excluderanges/hg19.UCSC.heterochromatin.rds",
                "excluderanges/hg19.UCSC.clone.rds",
                "excluderanges/hg19.UCSC.contig.rds",
                "excluderanges/hg19.UCSC.scaffold.rds",
                "excluderanges/mm39.UCSC.centromere.rds",
                "excluderanges/mm39.UCSC.telomere.rds",
                "excluderanges/mm39.UCSC.short_arm.rds",
                "excluderanges/mm39.UCSC.contig.rds",
                "excluderanges/mm39.UCSC.scaffold.rds",
                "excluderanges/mm10.UCSC.centromere.rds",
                "excluderanges/mm10.UCSC.telomere.rds",
                "excluderanges/mm10.UCSC.short_arm.rds",
                "excluderanges/mm10.UCSC.clone.rds",
                "excluderanges/mm10.UCSC.contig.rds",
                "excluderanges/mm10.UCSC.scaffold.rds",
                "excluderanges/mm10.UCSC.other.rds",
                "excluderanges/mm10.UCSC.fragment.rds",
                "excluderanges/mm9.UCSC.centromere.rds",
                "excluderanges/mm9.UCSC.fragment.rds",
                "excluderanges/mm9.UCSC.contig.rds",
                "excluderanges/danRer10.UCSC.contig.rds",
                "excluderanges/danRer10.UCSC.scaffold.rds",
                "excluderanges/dm6.UCSC.other.rds",
                "excluderanges/dm3.UCSC.contig.rds",
                "excluderanges/dm3.UCSC.scaffold.rds",
                "excluderanges/TAIR10.UCSC.araTha1.gap.rds"),
  Tags = c("T2T;excluderanges",
           "hg38;Kundaje;excluderanges",
           "hg38;Bernstein;excluderanges",
           "hg38;Boyle;excluderanges",
           "hg38;Kundaje;excluderanges",
           "hg38;Lareau;excluderanges",
           "hg38;Reddy;excluderanges",
           "hg38;Wimberley;excluderanges",
           "hg38;Wold;excluderanges",
           "hg38;Yeo;excluderanges",
           "hg19;Boyle;hg19-excluderanges",
           "hg19;Bernstein;excluderanges",
           "hg19;Birney;excluderanges",
           "hg19;Crawford;excluderanges",
           "hg19;Lareau;excluderanges",
           "hg19;Wold;excluderanges",
           "hg19;Yeo;excluderanges",
           "mm39;excluderanges",
           "mm10;Boyle;excluderanges",
           "mm10;Hardison;excluderanges",
           "mm10;Hardison;excluderanges",
           "mm10;Kundaje;excluderanges",
           "mm10;Kundaje;excluderanges",
           "mm10;Lareau;excluderanges",
           "mm10;Wold;excluderanges",
           "mm9;Lareau;excluderanges",
           "mm9;Wold;excluderanges",
           "ce11;Boyle;excluderanges",
           "ce10;Boyle;excluderanges",
           "ce10;Kundaje;excluderanges",
           "danRer10;Domingues;excluderanges",
           "danRer10;Yang;excluderanges",
           "dm6;Boyle;excluderanges",
           "dm3;Boyle;excluderanges",
           "dm3;Kundaje;excluderanges",
           "TAIR10;Wimberley;excluderanges",
           "TAIR10;Klasfeld;aexcluderanges",
           "TAIR10;Klasfeld;excluderanges",
           "T2T;Lareau;NUMTs",
           "hg38;Lareau;NUMTs",
           "hg19;Lareau;NUMTs",
           "mm10;Lareau;NUMTs",
           "mm9;Lareau;NUMTs",
           "hg19;UCSC;NUMTs",
           "mm9;UCSC;NUMTs",
           "T2T;CHM13;centromere",
           "T2T;CHM13;telomere",
           "T2T;UCSC;censat",
           "T2T;UCSC;gap",
           "T2T;UCSC;hgUnique",
           "hg38;UCSC;centromere",
           "hg38;UCSC;telomere",
           "hg38;UCSC;short_arm",
           "hg38;UCSC;heterochromatin",
           "hg38;UCSC;contig",
           "hg38;UCSC;scaffold",
           "hg19;UCSC;centromere",
           "hg19;UCSC;telomere",
           "hg19;UCSC;short_arm",
           "hg19;UCSC;heterochromatin",
           "hg19;UCSC;clone",
           "hg19;UCSC;contig",
           "hg19;UCSC;scaffold",
           "mm39;UCSC;centromere",
           "mm39;UCSC;telomere",
           "mm39;UCSC;short_arm",
           "mm39;UCSC;contig",
           "mm39;UCSC;scaffold",
           "mm10;UCSC;centromere",
           "mm10;UCSC;telomere",
           "mm10;UCSC;short_arm",
           "mm10;UCSC;clone",
           "mm10;UCSC;contig",
           "mm10;UCSC;scaffold",
           "mm10;UCSC;other",
           "mm10;UCSC;fragment",
           "mm9;UCSC;centromere",
           "mm9;UCSC;fragment",
           "mm9;UCSC;contig",
           "danRer10;UCSC;contig",
           "danRer10;UCSC;scaffold",
           "dm6;UCSC;other",
           "dm3;UCSC;contig",
           "dm3;UCSC;scaffold",
           "TAIR10;UCSC;gap"),
  Notes = "")


write.csv(meta, file="inst/extdata/metadata.csv", row.names=FALSE)
